github/src/main/kotlin/gropius/sync/github/Incoming.kt

```diff
@@ -105,27 +128,128 @@ class Incoming(
      * Sync github to gropius
      */
     suspend fun sync() {
-        val issueGrabber = IssueGrabber(repositoryInfoRepository, mongoOperations)
-        issueGrabber.requestNewNodes()
-        issueGrabber.iterate {
-            issueModified(it)
+        imsConfigManager.findTemplates()
+        for (imsTemplate in imsConfigManager.findTemplates()) {
+            for (ims in imsTemplate.usedIn()) {
+                try {
+                    val imsConfig = IMSConfig(helper, ims)
+                    syncIMS(imsConfig)
+                } catch (e: SyncNotificator.NotificatedError) {
+                    syncNotificator.sendNotification(
+                        ims, SyncNotificator.NotificationDummy(e)
+                    )
+                } catch (e: Exception) {
+                    logger.warn("Error in global sync", e)
+                }
+            }
         }
-        for (issue in issueInfoRepository.findByDirtyIsTrue().toList()) {
-            val timelineGrabber = TimelineGrabber(issueInfoRepository, mongoOperations, issue.githubId)
-            timelineGrabber.requestNewNodes()
+    }
+
+    /**
+     * Request an user token from the auth service
+     * @param ims IMS the token is requested for
+     * @param gropiusUser The user the token should be for
+     * @return token if available
+     */
+    suspend fun getGithubUserToken(ims: IMS, gropiusUser: GropiusUser): String? {
```

Comment:
maybe some of the new functions in this class can be private (or do you plan to use them in other classes too?)

Reply:
This function will be replaced in a PR VERY soon. Do not know anything about it yet